plugins/modules/rulebook_activation.py

```diff
@@ -518,20 +616,69 @@ def main() -> None:
         except EDAError as e:
             module.fail_json(msg=f"Failed to delete rulebook activation: {e}")
 
-    if activation:
-        module.exit_json(
-            msg=f"A rulebook activation with name: {name} already exists. "
-            "The module does not support modifying a rulebook activation.",
-            changed=False,
-            id=activation["id"],
-        )
-
     # Activation Data that will be sent for create/update
     activation_params = create_params(module, controller, is_aap_24=is_aap_24)
-    activation_params["name"] = name
+    activation_params["name"] = new_name if new_name else name
+
+    # Handle copies before anything else
```

Comment:
I would suggest creating two dedicated functions for these two new features and calling them in the main to keep the main function shorter. It will also be useful to understand if any of these can be moved into module_utils directly.